cogs/utils/generic.py

```diff
@@ -65,25 +65,22 @@ async def get_channels(self, ctx):
         await ctx.channel.send(f"```{output}```")
 
 
-    # Send a message to a user on their birth day
+    # Do stuff when a message is deleted
+    @commands.Cog.listener()
+    async def on_message_delete(self, message):
+        pass
+
+
+    # Send a message to a user on their birthday
     @tasks.loop(hours=24)
     async def birthday_alert(self):
-        # List to save id of members who already recieved a msg
-        member_ids_sent = []
-        guild_ids = await get_all_guilds()
-        for guild_id in guild_ids:
-            members = await get_all_members(guild_id)
-            for member in members:
-                member_id = member['member_id']
-                if member['birthday'] == None or member_id in member_ids_sent:
-                    continue
-                bday = datetime.strptime(member['birthday'], '%Y-%m-%d')
-                now = date.today()
-                if (bday.month, bday.day) == (now.month, now.day):
-                    user = self.bot.get_user(member_id)
-                    reply = f"Happy birthday, **{user.name}**!"
-                    await user.send(content=reply)
-                    member_ids_sent.append(member_id)
+        members = await get_members_bday_equals_date(date.today())
+        for member in members:
+            member_id = member['member_id']
+            user = self.bot.get_user(member_id)
+            reply = f"Happy birthday, **{user.name}**!"
+            await user.send(content=reply)
```

Comment:
Awesome, love the use of passing in the date.today() to even further reduce complexity of the search and avoid unnecessary steps!